src/overlays/actors/ovl_Obj_Moon_Stone/z_obj_moon_stone.c

```diff
@@ -21,28 +35,136 @@ const ActorInit Obj_Moon_Stone_InitVars = {
     (ActorFunc)ObjMoonStone_Update,
     (ActorFunc)ObjMoonStone_Draw,
 };
-*/
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/ObjMoonStone_Init.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/ObjMoonStone_Destroy.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C0662C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C06640.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C066F8.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C0670C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C0673C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C06768.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C0685C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/func_80C06870.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/ObjMoonStone_Update.asm")
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Moon_Stone_0x80C06510/ObjMoonStone_Draw.asm")
+extern AnimatedMaterial D_06001C60;
+extern Gfx D_06000D78[];
+extern Gfx D_06000C80[];
+
+void ObjMoonStone_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjMoonStone* this = THIS;
+
+    Actor_SetScale(this, 0.3f);
+    this->unk194 = ((this->actor.params & 0xF000) >> 0xC);
```

Comment:
Normally the params bitwise access need a macro. You did not start documentation on this file, so it will not be required.